src/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/CompilerServices/ProjectData.vb

```diff
@@ -34,5 +33,88 @@ Namespace Global.Microsoft.VisualBasic.CompilerServices
                 m_oProject = GetProjectData
             End If
         End Function
+
+        ''' <summary>
+        ''' This function is called by the compiler in response to err code, e.g. err 123
+        ''' It is also called when the compiler encounters a resume that isn't preceded by an On Error command
+        ''' </summary>
+        ''' <param name="hr"></param>
+        ''' <returns></returns>
+        ''' <remarks></remarks>
+        Public Shared Function CreateProjectError(ByVal hr As Integer) As System.Exception
+            '*************************
+            '*** PERFORMANCE NOTE: ***
+            '*************************
+            ' Err Object is <ThreadStatic> and is pretty expensive to access so we cache to a local to cut the number of accesses
+            Dim ErrObj As ErrObject = Err()
+            ErrObj.Clear()
+            Dim ErrNumber As Integer = ErrObj.MapErrorNumber(hr)
+            Return ErrObj.CreateException(hr, Utils.GetResourceString(CType(ErrNumber, vbErrors)))
+        End Function
+
+        ''' <summary>
+        ''' Called by the compiler in response to falling into a catch block.
+        ''' Inside the catch statement the compiler generates code to call:
+        '''  ProjectData::SetProjectError(exception)  That call
+        '''  in turns sets the ErrObject which is accessed via the VB Err statement.
+        '''  So a VB6 programmer would typically then do something like:
+        '''    if err.Number = * do something where err accesses the ErrObject that
+        '''  is set by this method.
+        ''' </summary>
+        ''' <param name="ex"></param>
+        ''' <remarks></remarks>
+        <SecuritySafeCritical()>
+        <ReliabilityContract(Consistency.WillNotCorruptState, Cer.Success)>
+        Public Overloads Shared Sub SetProjectError(ByVal ex As Exception)
+            ' The Try/Finally and constrained regions calls guarantee success under high
+            ' stress conditions by enabling eager jitting of the finally block
+            System.Runtime.CompilerServices.RuntimeHelpers.PrepareConstrainedRegions()
```

Comment:
@jkotas, I don't remember... do we still eagerly JIT finally blocks with PrepareConstrainedRegions in core?

Reply:
Constrained execution regions are not supported in net core. Methods like PrepareConstrainedRegions are no-ops. They can be deleted.